core-java-modules/core-java-lang-2/src/test/java/com/baeldung/headlessmode/HeadlessModeUnitTest.java

```diff
@@ -0,0 +1,83 @@
+package com.baeldung.headlessmode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import java.awt.Canvas;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Frame;
+import java.awt.GraphicsEnvironment;
+import java.awt.HeadlessException;
+import java.awt.image.BufferedImage;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import javax.imageio.ImageIO;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HeadlessModeUnitTest {
+
+    private static final String IN_FILE = "src/test/resources/product.png";

```

Comment:
it sounds to me this is a good candidate to get the stream as a resource rather than via filesystem, what do you think?

Reply:
I can get the stream as resource, but for the write operation i still need the whole path. So to keep in and out file code consistent, i used this way. If you still insist, i can change the in-file code though :)

Reply:
I see! the best option is to read as resource and write to a temp folder, in this way we are not dependent from the way the source code is structured nor the building system we are using.

Reply:
Ok, that makes sense. I've updated the code.